python/ray/train/tests/test_jax.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+import ray
+from ray.air.config import ScalingConfig
+from ray.train.constants import TRAINING_ITERATION
+from ray.train.examples.jax_mnist_example import (
+    train_func as jax_mnist_train_func,
+)
+from ray.train.jax import JaxTrainer
+
+from ray import tune
+from ray.tune.tune_config import TuneConfig
+from ray.tune.tuner import Tuner
+
+
+@pytest.fixture
+def ray_start_8_cpus():
+    address_info = ray.init(num_cpus=8)
+    yield address_info
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+def test_jax_mnist_gpu(ray_start_8_cpus):
+    num_workers = 1
+    num_epochs = 2
+    trainer = JaxTrainer(
+        train_loop_per_worker=jax_mnist_train_func,
+        train_loop_config={
+            "num_epochs": num_epochs,
+            "learning_rate": 1e-3,
+            "momentum": 0.9,
+            "batch_size": 8192,
+        },
+        scaling_config=ScalingConfig(
+            num_workers=num_workers,
+            use_gpu=False,
+        ),
+    )
+
+    results = trainer.fit()
+    result = results.metrics
+    assert result[TRAINING_ITERATION] == num_epochs
+
+
+def tune_jax_mnist(num_workers, use_gpu, num_samples, num_gpus_per_worker=0):
+    trainer = JaxTrainer(
+        jax_mnist_train_func,
+        scaling_config=ScalingConfig(
+            num_workers=num_workers,
+            use_gpu=use_gpu,
+            resources_per_worker={"GPU": num_gpus_per_worker} if use_gpu else None,
+        ),
+    )
+    tuner = Tuner(
+        trainer,
+        param_space={
+            "train_loop_config": {
+                "lr": tune.loguniform(1e-4, 1e-1),
+                "batch_size": tune.choice([32, 64, 128]),
+                "epochs": 2,
+            }
+        },
+        tune_config=TuneConfig(
+            num_samples=num_samples,
+        ),
+    )
+    analysis = tuner.fit()._experiment_analysis
+
+    # Check that loss decreases in each trial.
+    for path, df in analysis.trial_dataframes.items():
+        assert df.loc[1, "loss"] < df.loc[0, "loss"]
+
+
+def test_tune_jax_mnist(ray_start_8_cpus):
+    tune_jax_mnist(num_workers=2, use_gpu=False, num_samples=2)
+
```

Comment:
can we add unit tests for `_validate_scaling_config`?

Reply:
thanks @Yard1 good idea, let me think about this!

Reply:
sounds good, added!